test/sparse/sparse.jl

```diff
@@ -1,469 +1,518 @@
+            @test f(arr, (1, 2)) ≈ [f(farr)]
+            @test isequal(f(arr, 3), f(farr, 3))
+        end
     end
 
-    # case where f(0) would throw
-    @test f(x->sqrt(x-1), pA+1) ≈ f(sqrt.(pA))
-    # these actually throw due to #10533
-    # @test f(x->sqrt(x-1), pA+1, 1) ≈ f(sqrt(pA), 1)
-    # @test f(x->sqrt(x-1), pA+1, 2) ≈ f(sqrt(pA), 2)
-    # @test f(x->sqrt(x-1), pA+1, 3) ≈ f(pA)
-end
-
-# empty cases
-@test sum(sparse(Int[])) === 0
-@test prod(sparse(Int[])) === 1
-@test_throws ArgumentError minimum(sparse(Int[]))
-@test_throws ArgumentError maximum(sparse(Int[]))
-@test var(sparse(Int[])) === NaN
+    for f in (sum, prod, minimum, maximum)
+        # Test with a map function that maps to non-zero
+        for arr in (se33, sA, pA)
+            @test f(x->x+1, arr) ≈ f(arr+1)
+        end
 
-for f in (sum, prod, minimum, maximum, var)
-    @test isequal(f(spzeros(0, 1), 1), f(Array{Int}(0, 1), 1))
-    @test isequal(f(spzeros(0, 1), 2), f(Array{Int}(0, 1), 2))
-    @test isequal(f(spzeros(0, 1), (1, 2)), f(Array{Int}(0, 1), (1, 2)))
-    @test isequal(f(spzeros(0, 1), 3), f(Array{Int}(0, 1), 3))
+        # case where f(0) would throw
+        @test f(x->sqrt(x-1), pA+1) ≈ f(sqrt.(pA))
+        # these actually throw due to #10533
+        # @test f(x->sqrt(x-1), pA+1, 1) ≈ f(sqrt(pA), 1)
+        # @test f(x->sqrt(x-1), pA+1, 2) ≈ f(sqrt(pA), 2)
+        # @test f(x->sqrt(x-1), pA+1, 3) ≈ f(pA)
+    end
 end
 
-# spdiagm
-@test Array(spdiagm((ones(2), ones(2)), (0, -1), 3, 3)) ==
-                       [1.0  0.0  0.0; 1.0  1.0  0.0;  0.0  1.0  0.0]
-@test Array(spdiagm(ones(2), -1, 3, 3)) == diagm(ones(2), -1)
+@testset "Empty cases" begin
+    @test sum(sparse(Int[])) === 0
+    @test prod(sparse(Int[])) === 1
+    @test_throws ArgumentError minimum(sparse(Int[]))
+    @test_throws ArgumentError maximum(sparse(Int[]))
+    @test var(sparse(Int[])) === NaN
 
-# issue #4986, reinterpret
-sfe22 = speye(Float64, 2)
-mfe22 = eye(Float64, 2)
-@test reinterpret(Int64, sfe22) == reinterpret(Int64, mfe22)
+    for f in (sum, prod, minimum, maximum, var)
+        @test isequal(f(spzeros(0, 1), 1), f(Array{Int}(0, 1), 1))
+        @test isequal(f(spzeros(0, 1), 2), f(Array{Int}(0, 1), 2))
+        @test isequal(f(spzeros(0, 1), (1, 2)), f(Array{Int}(0, 1), (1, 2)))
+        @test isequal(f(spzeros(0, 1), 3), f(Array{Int}(0, 1), 3))
+    end
+end
 
-# issue #5190
-@test_throws ArgumentError sparsevec([3,5,7],[0.1,0.0,3.2],4)
+@testset "Sparse Diagonal Matrix" begin
+    @test Array(spdiagm((ones(2), ones(2)), (0, -1), 3, 3)) ==
+                           [1.0  0.0  0.0; 1.0  1.0  0.0;  0.0  1.0  0.0]
+    @test Array(spdiagm(ones(2), -1, 3, 3)) == diagm(ones(2), -1)
+end
 
+@testset "Issue #4986, reinterpret" begin
+    sfe22 = speye(Float64, 2)
+    mfe22 = eye(Float64, 2)
+    @test reinterpret(Int64, sfe22) == reinterpret(Int64, mfe22)
+end
 
-# issue #5386
-K,J,V = findnz(SparseMatrixCSC(2,1,[1,3],[1,2],[1.0,0.0]))
```

Comment:
Perhaps expand to something along the lines of "Issue described in https://..."?